File: C++/Paths from root with a specified sum.cpp
class Solution
{
    public:
    vector<vector<int>> result;
    vector<int> ans;
    void pathfinder(Node *root,int curr_sum,int sum)
    {
        if(root==NULL)
            return;
        curr_sum+=root->key;
        ans.push_back(root->key);
        if(curr_sum==sum)
            result.push_back(ans);
        pathfinder(root->left,curr_sum,sum);
        pathfinder(root->right,curr_sum,sum);
        ans.pop_back();
        return;
    }
    vector<vector<int>> printPaths(Node *root, int sum)
    {
        pathfinder(root,0,sum);
        return result;
    }
};

File: JAVA/Paths from root with a specified sum.java
class Solution
{
    
    public static void helper(ArrayList<ArrayList<Integer>> list,Node root,int sum,ArrayList<Integer>temp,int target){
        if(root==null) return;
        
        sum+=root.data;
        temp.add(root.data);
        if(sum==target){
            list.add(new ArrayList<>(temp));
        }
        helper(list,root.left,sum,temp,target);
        helper(list,root.right,sum,temp,target);
        
        sum-=root.data;
        temp.remove(temp.size()-1);
        
    }
    public static ArrayList<ArrayList<Integer>> printPaths(Node root, int sum)
    {
        // code here
        ArrayList<ArrayList<Integer>> list = new ArrayList<>();
        helper(list,root,0,new ArrayList<>(),sum);

        return list;
    }
}
